src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+                CAmount value = std::get<1>(smr);
+                std::string hexMemo = std::get<2>(smr);
+                zOutputsDeque.pop_front();
+
+                PaymentAddress pa = CZCPaymentAddress(address).Get();
+                JSOutput jso = JSOutput(pa, value);
+                if (hexMemo.size() > 0) {
+                    jso.memo = get_memo_from_hex_string(hexMemo);
+                }
+                info.vjsout.push_back(jso);
+                
+                // Funds are removed from the value pool and enter the private pool
+                info.vpub_old += value;
+            }
+            obj = perform_joinsplit(info);
+        }
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #2
+     */   
+ 
+    
+    
+    /**
+     * SCENARIO #3
+     * 
+     * zaddr -> taddrs
+     *       -> zaddrs
+     * 
+     * Processing order:
+     * Part 1: taddrs and miners fee
+     * Part 2: zaddrs 
+     */
+    
+    /**
+     * SCENARIO #3
+     * Part 1: Add to the transparent value pool.
+     */
+    Object obj;
+    CAmount jsChange = 0;   // this is updated after each joinsplit
+    bool minersFeeProcessed = false;
+
+    if (t_outputs_total > 0) {
+        add_taddr_outputs_to_tx();
+        CAmount taddrTargetAmount = t_outputs_total + minersFee;
+        minersFeeProcessed  = true;
+        while (zInputsDeque.size() > 0 && taddrTargetAmount > 0) {
+            AsyncJoinSplitInfo info;
+            info.vpub_old = 0;
+            info.vpub_new = 0;
+            std::vector<JSOutPoint> outPoints;
+            int n = 0;
+            while (n++ < 2 && taddrTargetAmount > 0) {
+                SendManyInputJSOP o = zInputsDeque.front();
+                JSOutPoint outPoint = std::get<0>(o);
+                Note note = std::get<1>(o);
+                CAmount noteFunds = std::get<2>(o);
+                zInputsDeque.pop_front();
+
+                info.notes.push_back(note);
+                outPoints.push_back(outPoint);
+
+
+                LogPrint("asyncrpc", "%s: spending note (txid=%s, vjoinsplit=%d, ciphertext=%d, amount=%s)\n",
+                        getId().substr(0, 10),
+                        outPoint.hash.ToString().substr(0, 10),
+                        outPoint.js,
+                        int(outPoint.n), // uint8_t
+                        FormatMoney(noteFunds, false)
+                        );
+
+                
+                // Put value back into the value pool
+                if (noteFunds >= taddrTargetAmount) {
+                    jsChange = noteFunds - taddrTargetAmount;
+                    info.vpub_new += taddrTargetAmount;
+                } else {
+                    info.vpub_new += noteFunds;
+                }
+
+                taddrTargetAmount -= noteFunds;
+                if (taddrTargetAmount <= 0) {
+                    break;
+                }
+            }
+
+            if (jsChange > 0) {
```

Comment:
If this is not the first iteration of the outer while loop, couldn't ``jsChange`` still be set from a previous iteration?